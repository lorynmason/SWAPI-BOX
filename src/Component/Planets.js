import React from 'react'
import './styles/main.scss'
import PropTypes from 'prop-types'

const Planets = ({ planets, toggleFavorites }) => {
  const cards = planets.map((planet) => {
    let resident
    if (planet.residents.length === 0) {
      resident = 'NO KNOWN OCCUPANTS'
    } else {
      resident = planet.residents.map((resident) => {
        return resident + ' | '
      })
    }
    return (
      <div className="planet-card card">
        <button className="favorite-btn" type="button" onClick={() => toggleFavorites(planet.id)}>
          <i className="fas fa-jedi" />
        </button>
        <h1>
          {planet.planet}
        </h1>
        <p>
          {`Climate: `}
          {planet.climate}
        </p>
        <p>
          {`Population: `}
          {planet.population}
        </p>
        <p>
          {`Residents: `}
          {resident}
        </p>
        <p>
          {`Terrain: `}
          {planet.terrain}
        </p>
      </div>
    )
  })

  return (
    <section className="cardContainer">
      {cards}
    </section>
  )
}

Planets.propTypes = {
  planets: PropTypes.array.isRequired
}

export default Planets
