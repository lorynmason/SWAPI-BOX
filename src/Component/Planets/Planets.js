import React from 'react'
import '../styles/main.scss'
import PropTypes from 'prop-types'

const Planets = ({ planets, toggleFavorites }) => {
  const cards = planets.map((planet) => {
    const infoValue = Object.values(planet.info).map((plInfo) => {
      return (
        <p>{plInfo}</p>
      )
    })

    return (
      <div className="planet-card card">
        <button className="favorite-btn" type="button" onClick={() => toggleFavorites(planet.id)}>
          <i className="fas fa-jedi" />
        </button>
        <div className="info">
          <p>
            {infoValue}
          </p>
        </div>
      </div>
    )
  })

  return (
    <section className="cardContainer">
      {cards}
    </section>
  )
}

Planets.propTypes = {
  planets: PropTypes.array.isRequired,
  toggleFavorites: PropTypes.func.isRequired
}

export default Planets
