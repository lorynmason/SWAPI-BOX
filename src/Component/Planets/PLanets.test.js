import React from 'react'
import { shallow } from 'enzyme'
import Planets from './Planets';

describe('Plantes', () => {
  const toggleFavorites = jest.fn()
  let wrapper;
  let mockPlanetData = [
    {
      info: {
      Climate: "Climate:  temperate",
      Name: "Alderaan",
      Population: "Population: 2000000000",
      Residents: "Residents:  Leia Organa,Bail Prestor Organa,Raymus Antilles",
      Terrain: "Terrain:  grasslands, mountains",
      }
    }
  ]
  beforeEach(() => {
    wrapper = shallow(<Planets planets={mockPlanetData} toggleFavorites={toggleFavorites}/>)
  })

  it( 'should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('favorite button should call toggleFavorites on click', () => {
    const button = wrapper.find('button')
    button.simulate('click')

    expect(toggleFavorites).toHaveBeenCalled()
  })
})