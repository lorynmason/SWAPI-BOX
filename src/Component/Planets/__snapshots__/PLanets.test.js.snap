// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plantes should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Planets
    planets={
      Array [
        Object {
          "category": "planets",
          "id": "Alderaan",
          "info": Object {
            "Climate": "Climate:  temperate",
            "Name": "Alderaan",
            "Population": "Population: 2000000000",
            "Residents": "Residents:  Leia Organa,Bail Prestor Organa,Raymus Antilles",
            "Terrain": "Terrain:  grasslands, mountains",
          },
        },
      ]
    }
    toggleFavorites={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="planet-card card"
        >
          <button
            className="favorite-btn"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-jedi"
            />
          </button>
          <div
            className="info"
          >
            <p>
              <p>
                Climate:  temperate
              </p>
              <p>
                Alderaan
              </p>
              <p>
                Population: 2000000000
              </p>
              <p>
                Residents:  Leia Organa,Bail Prestor Organa,Raymus Antilles
              </p>
              <p>
                Terrain:  grasslands, mountains
              </p>
            </p>
          </div>
        </div>,
      ],
      "className": "cardContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Alderaan-0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="favorite-btn"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-jedi"
              />
            </button>,
            <div
              className="info"
            >
              <p>
                <p>
                  Climate:  temperate
                </p>
                <p>
                  Alderaan
                </p>
                <p>
                  Population: 2000000000
                </p>
                <p>
                  Residents:  Leia Organa,Bail Prestor Organa,Raymus Antilles
                </p>
                <p>
                  Terrain:  grasslands, mountains
                </p>
              </p>
            </div>,
          ],
          "className": "planet-card card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-jedi"
              />,
              "className": "favorite-btn",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-jedi",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                <p>
                  Climate:  temperate
                </p>
                <p>
                  Alderaan
                </p>
                <p>
                  Population: 2000000000
                </p>
                <p>
                  Residents:  Leia Organa,Bail Prestor Organa,Raymus Antilles
                </p>
                <p>
                  Terrain:  grasslands, mountains
                </p>
              </p>,
              "className": "info",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Climate:  temperate
                  </p>,
                  <p>
                    Alderaan
                  </p>,
                  <p>
                    Population: 2000000000
                  </p>,
                  <p>
                    Residents:  Leia Organa,Bail Prestor Organa,Raymus Antilles
                  </p>,
                  <p>
                    Terrain:  grasslands, mountains
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "C-0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Climate:  temperate",
                  },
                  "ref": null,
                  "rendered": "Climate:  temperate",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": "A-1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Alderaan",
                  },
                  "ref": null,
                  "rendered": "Alderaan",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": "P-2",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Population: 2000000000",
                  },
                  "ref": null,
                  "rendered": "Population: 2000000000",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": "R-3",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Residents:  Leia Organa,Bail Prestor Organa,Raymus Antilles",
                  },
                  "ref": null,
                  "rendered": "Residents:  Leia Organa,Bail Prestor Organa,Raymus Antilles",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": "T-4",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Terrain:  grasslands, mountains",
                  },
                  "ref": null,
                  "rendered": "Terrain:  grasslands, mountains",
                  "type": "p",
                },
              ],
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="planet-card card"
          >
            <button
              className="favorite-btn"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-jedi"
              />
            </button>
            <div
              className="info"
            >
              <p>
                <p>
                  Climate:  temperate
                </p>
                <p>
                  Alderaan
                </p>
                <p>
                  Population: 2000000000
                </p>
                <p>
                  Residents:  Leia Organa,Bail Prestor Organa,Raymus Antilles
                </p>
                <p>
                  Terrain:  grasslands, mountains
                </p>
              </p>
            </div>
          </div>,
        ],
        "className": "cardContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Alderaan-0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="favorite-btn"
                onClick={[Function]}
                type="button"
              >
                <i
                  className="fas fa-jedi"
                />
              </button>,
              <div
                className="info"
              >
                <p>
                  <p>
                    Climate:  temperate
                  </p>
                  <p>
                    Alderaan
                  </p>
                  <p>
                    Population: 2000000000
                  </p>
                  <p>
                    Residents:  Leia Organa,Bail Prestor Organa,Raymus Antilles
                  </p>
                  <p>
                    Terrain:  grasslands, mountains
                  </p>
                </p>
              </div>,
            ],
            "className": "planet-card card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-jedi"
                />,
                "className": "favorite-btn",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-jedi",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  <p>
                    Climate:  temperate
                  </p>
                  <p>
                    Alderaan
                  </p>
                  <p>
                    Population: 2000000000
                  </p>
                  <p>
                    Residents:  Leia Organa,Bail Prestor Organa,Raymus Antilles
                  </p>
                  <p>
                    Terrain:  grasslands, mountains
                  </p>
                </p>,
                "className": "info",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      Climate:  temperate
                    </p>,
                    <p>
                      Alderaan
                    </p>,
                    <p>
                      Population: 2000000000
                    </p>,
                    <p>
                      Residents:  Leia Organa,Bail Prestor Organa,Raymus Antilles
                    </p>,
                    <p>
                      Terrain:  grasslands, mountains
                    </p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "C-0",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Climate:  temperate",
                    },
                    "ref": null,
                    "rendered": "Climate:  temperate",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": "A-1",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Alderaan",
                    },
                    "ref": null,
                    "rendered": "Alderaan",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": "P-2",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Population: 2000000000",
                    },
                    "ref": null,
                    "rendered": "Population: 2000000000",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": "R-3",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Residents:  Leia Organa,Bail Prestor Organa,Raymus Antilles",
                    },
                    "ref": null,
                    "rendered": "Residents:  Leia Organa,Bail Prestor Organa,Raymus Antilles",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": "T-4",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Terrain:  grasslands, mountains",
                    },
                    "ref": null,
                    "rendered": "Terrain:  grasslands, mountains",
                    "type": "p",
                  },
                ],
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
